tests/apache/livy/triggers/test_livy.py

```diff
@@ -0,0 +1,175 @@
+    task = [i async for i in trigger.run()]
+    assert len(task) == 1
+    assert (
+        TriggerEvent({"status": "success", "batch_id": 1, "response": "Batch 1 succeeded", "log_lines": None})
+        in task
+    )
+
+
+@pytest.mark.asyncio
+@mock.patch("astronomer.providers.apache.livy.triggers.livy.LivyTrigger.poll_for_termination")
+async def test_livy_trigger_run_with_poll_interval_success(mock_poll_for_termination):
+    """
+    Test if the task is run is in triggerr successfully with poll interval=30.
+    :return:
+    """
+    mock_poll_for_termination.return_value = {"status": "success"}
+    trigger = LivyTrigger(batch_id=1, spark_params={}, livy_conn_id="livy_default", polling_interval=30)
+
+    task = [i async for i in trigger.run()]
+    assert len(task) == 1
+    assert TriggerEvent({"status": "success"}) in task
+
+
+@pytest.mark.asyncio
+@mock.patch("astronomer.providers.apache.livy.triggers.livy.LivyTrigger.poll_for_termination")
+async def test_livy_trigger_run_with_poll_interval_error(mock_poll_for_termination):
+    """
+    Test if the task is run is in triggerr failed with poll interval=30.
+    :return:
+    """
+    mock_poll_for_termination.return_value = {"status": "error"}
+    trigger = LivyTrigger(batch_id=1, spark_params={}, livy_conn_id="livy_default", polling_interval=30)
+
+    task = [i async for i in trigger.run()]
+    assert len(task) == 1
+    assert TriggerEvent({"status": "error"}) in task
+
+
+@pytest.mark.asyncio
+async def test_livy_trigger_run_with_exception():
+    """
+    Test if the task is run is in triggerr failed with error message.
+    :return:
+    """
+    trigger = LivyTrigger(batch_id=1, spark_params={}, livy_conn_id="livy_default", polling_interval=30)
+
+    task = [i async for i in trigger.run()]
+    assert len(task) == 1
+    assert (
+        TriggerEvent(
+            {
+                "status": "error",
+                "batch_id": 1,
+                "response": "Batch 1 did not succeed with Cannot connect to host livy:8998 ssl:default "
+                "[Name or service not known]",
+                "log_lines": None,
+            }
+        )
+        in task
+    )
+
+
+@pytest.mark.asyncio
+async def test_livy_trigger_poll_for_termination_with_client_error():
+    """
+    Test if the poll_for_termination polls for batch with client error
+    :return:
+    """
+    trigger = LivyTrigger(batch_id=1, spark_params={}, livy_conn_id="livy_default", polling_interval=30)
+
+    with pytest.raises(ClientConnectionError):
+        await trigger.poll_for_termination(1)
+
+
+@pytest.mark.asyncio
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.get_batch_state")
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.dump_batch_logs")
+async def test_livy_trigger_poll_for_termination_success(mock_dump_batch_logs, mock_get_batch_state):
+    """
+    Test if the poll_for_termination polls for batch to terminate for success
+    :return:
```

Comment:
?

Reply:
`Test if the poll_for_termination() in the triggerer returned success reponse when get_batch_state() runs successfully.`

Changing it in the docstring.

Reply:
[Done](https://github.com/astronomer/astronomer-providers/pull/178/commits/132fa197b35937edf9d07a1eac1d94a0350248fd)